### PR TITLE
Prepare 0.12.0 release

### PR DESCRIPTION
**Pull request by mtreinish**
This commit prepares the 0.12.0 release. It moves the release notes that for the release to a separate directory and adds a prelude to the release notes. This should be the last commit to merge and after it merges it'll be tagged for the 0.12.0 release.

TODO:

- [x] Wait for all PRs to merge and issues to close
- [x] Review and update release notes content

### CODE REVIEW

**Comment by mtreinish** [comment]
I think this is ready (assuming the gha outage https://www.githubstatus.com/incidents/smn1qtqvbsb6 ends soon and everything passes CI). Once this merges I think we're ready to tag and release 0.12.0

File: releasenotes/notes/0.12/prepare-0.12.0-45e978cfb17b78a5.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  manylinux2014 for all precompiled Linux binaries we publish to PyPI. The
+  minimum supported Rust version for building rustworkx from source has
+  increased to Rust 1.48.
+
+  This release also includes several new features, some highlights are:
+
+    * Support for graph attributes under the ``attrs`` attribute
+    * New serialization format support (a graphml parser, :func:`~.read_graphml`,
+      and a node link JSON generator, :func:`~.node_link_json`)
+    * New algorithms functions
```
Comment (IvanIsCoding) [inline releasenotes/notes/0.12/prepare-0.12.0-45e978cfb17b78a5.yaml:25]:
```suggestion
    * Eigen Vector Centrality
    * Stoer–Wagner Min-Cut algorithm
    * Bellman-Ford shortest path algorithm
```

**Review by IvanIsCoding — APPROVED** [review]
LGTM, I just made a minor suggestion to cite (in my personal opinion) the most popular algorithms we added

File: releasenotes/notes/0.12/prepare-0.12.0-45e978cfb17b78a5.yaml (hunk 1 above)
Comment (mtreinish) [reply releasenotes/notes/0.12/prepare-0.12.0-45e978cfb17b78a5.yaml:25]: What about something like:

```suggestion
    * New algorithms functions including:
        * Eigenvector Centrality
        * Stoer–Wagner Min-Cut algorithm
        * Bellman-Ford shortest path algorithm

```

**Pull request merged**